cmd/server/main.go

```diff
@@ -94,6 +94,15 @@ func main() {
 	}()
 
 	go func() {
+		// Server should unlink the socket file prior to binding it in case the socket isn't cleaned up gracefully.
+		// This can happen if the application is killed by SIGKILL or SIGSTOP, i.e. kill -9 or docker kill by default.
+		zap.L().Info("unlinking ", zap.String("port", *addr))
+		err = syscall.Unlink(*addr)
+		if err != nil {
+			if err.Error() != "no such file or directory" {
+				zap.L().Warn("Failed to unlink unix socket. %v", zap.Error(err))
+			}
```

Comment:
If you get `"no such file or directory"`, doesn't that mean the socket was correctly cleaned up and you didn't need to remove it in the first place? Why do you need to log a warning in that case?

Also, you're swallowing any other error, such as insufficient permission, which will fail later.